mava/evaluator.py

```diff
@@ -286,36 +286,54 @@ def evaluator_fn(
 
 def make_eval_fns(
     eval_env: Environment,
-    network: Any,
+    network_apply_fn: Union[ActorApply, RecActorApply],
     config: DictConfig,
     use_recurrent_net: bool = False,
     scanned_rnn: Optional[nn.Module] = None,
 ) -> Tuple[EvalFn, EvalFn]:
-    """Initialise evaluator_fn."""
+    """Initialize evaluator functions for reinforcement learning.
+
+    Args:
+        eval_env (Environment): The environment used for evaluation.
+        network_apply_fn (Union[ActorApply,RecActorApply]): Creates a policy to sample.
+        config (DictConfig): The configuration settings for the evaluation.
+        use_recurrent_net (bool, optional): Whether to use a rnn. Defaults to False.
+        scanned_rnn (Optional[nn.Module], optional): The rnn module.
+            Required if `use_recurrent_net` is True. Defaults to None.
+
+    Returns:
+        Tuple[EvalFn, EvalFn]: A tuple of two evaluation functions:
+        one for use during training and one for absolute metrics.
+
+    Raises:
+        AssertionError: If `use_recurrent_net` is True but `scanned_rnn` is not provided.
+    """
     # Check if win rate is required for evaluation.
     log_win_rate = config.env.eval_metric == "win_rate"
     # Vmap it over number of agents and create evaluator_fn.
     if use_recurrent_net:
         assert scanned_rnn is not None
         evaluator = get_rnn_evaluator_fn(
             eval_env,
-            network.apply,
+            network_apply_fn,  # type: ignore
```

Comment:
What is the mypy error we get when we don't type ignore here?

Reply:
It complains because get_rnn_evaluator_fn expects RecActorApply type there, but the input is Union[ActorApply, RecActorApply] - so it could be either type, which we need. For reference, before this PR we used the type "Any" for passing the network through, so it didn't make mypy angry.

Reply:
Makes total sense. I think we can just type ignore in that case.